### PR TITLE
allow for skipping of multiple tests

### PR DESCRIPTION
**Pull request by edsu7**
- removed file length check for `c609`
- added new CLI syntax for skipping tests:
`seq-tools validate tests/submissions/BRCA1234T.WXS.skip-md5/sequencing_experiment.json -k c683`(new) == `seq-tools validate tests/submissions/BRCA1234T.WXS.skip-md5/sequencing_experiment.json --skip-md5-check`(old)
- now allows for multiple test skipping:
`seq-tools validate tests/submissions/BRCA1234T.WXS.skip-md5/sequencing_experiment.json -k c683 -k c670`
- Perhaps to lenient? skipping critical dependency like `c605` results in :
`seq-tools validate tests/submissions/HCC1160T.valid/sequencing_experiment.json -d tests/seq-data/ -k c605`
```
(base)% cat validation_report.UNKNOWN.jsonl | jq
{
  "tool": {
    "name": "seq-tools",
    "version": "1.2.4"
  },
  "metadata_file": "/Users/esu/Desktop/GitHub/seq-tools/tests/submissions/HCC1160T.valid/sequencing_experiment.json",
  "data_dir": "/Users/esu/Desktop/GitHub/seq-tools/tests/seq-data",
  "started_at": "2023-02-06T19:25:25.466Z",
  "ended_at": "2023-02-06T19:25:25.709Z",
  "validation": {
    "status": "UNKNOWN",
    "message": "Please see individual checks for details",
    "checks": [
      {
        "checker": "c608_bam_sanity",
        "status": "UNKNOWN",
        "message": "Not all dependent checks are in VALID status: c605_all_files_accessible: SKIPPED. Please fix reported problem and then run the validation again."
      },
      {
        "checker": "c609_fastq_sanity",
        "status": "UNKNOWN",
        "message": "Not all dependent checks are in VALID status: c605_all_files_accessible: SKIPPED. Please fix reported problem and then run the validation again."
      },
      {
        "checker": "c660_metadata_in_bam_rg_header",
        "status": "UNKNOWN",
        "message": "Not all dependent checks are in VALID status: c605_all_files_accessible: SKIPPED. Please fix reported problem and then run the validation again."
      },
      {
        "checker": "c670_rg_is_paired_in_bam",
        "status": "UNKNOWN",
        "message": "Not all dependent checks are in VALID status: c608_bam_sanity: UNKNOWN, c608_bam_sanity: UNKNOWN. Please fix reported problem and then run the validation again."
      },
      {
        "checker": "c680_repeated_read_names_per_group_in_bam",
        "status": "UNKNOWN",
        "message": "Not all dependent checks are in VALID status: c605_all_files_accessible: SKIPPED, c608_bam_sanity: UNKNOWN, c670_rg_is_paired_in_bam: UNKNOWN. Please fix reported problem and then run the validation again."
      },
      {
        "checker": "c681_fileSize_match",
        "status": "UNKNOWN",
        "message": "Not all dependent checks are in VALID status: c605_all_files_accessible: SKIPPED. Please fix reported problem and then run the validation again."
      },
      {
        "checker": "c683_fileMd5sum_match",
        "status": "UNKNOWN",
        "message": "Not all dependent checks are in VALID status: c605_all_files_accessible: SKIPPED. Please fix reported problem and then run the validation again."
      }
    ]
  }
}
```

### CODE REVIEW

**Comment by lindaxiang** [comment]
@edsu7 , the skipping option looks good.
To avoid the scenario of skipping critical dependency like c605, we can have an allowed_list for skipped-checks. 
For example, let's narrow the list only contains c681 and c683 as we know these two checks will be captured in score uploading stage.

**Review by lindaxiang — APPROVED** [review]
Test it and lgtm.

**Comment by edsu7** [comment]
> @edsu7 , the skipping option looks good. To avoid the scenario of skipping critical dependency like c605, we can have an allowed_list for skipped-checks. For example, let's narrow the list only contains c681 and c683 as we know these two checks will be captured in score uploading stage.

Added the following code to address this:
https://github.com/icgc-argo/seq-tools/blob/35abd2393eb8a7641df729c7c59c438d6ed09736/seq_tools/validation/__init__.py#L51-L69

The following tests:
```
seq-tools validate tests/submissions/HCC1160T.valid/sequencing_experiment.json -d tests/seq-data/ -k c605                                                                        
Start validating 1 metadata file(s), current_time: 2023-02-06T22:18:24.317Z. Please be patient, it may take sometime.
c605 test is not skippable
Aborted!
```

**Comment by edsu7** [comment]
As per Linda's suggestion :
https://github.com/icgc-argo/seq-tools/pull/109#discussion_r1099324393

Removed:
https://github.com/icgc-argo/seq-tools/blob/35abd2393eb8a7641df729c7c59c438d6ed09736/seq_tools/validation/__init__.py#L51-L69

Added:
https://github.com/icgc-argo/seq-tools/blob/f3fc5ecd80bfd3e67e7cd1f69580d11877d8e3f4/seq_tools/cli/__init__.py#L60

File: seq_tools/cli/__init__.py [hunk 1]
```diff
@@ -55,13 +56,19 @@ def main(ctx, debug, ignore_update):
 @click.option('--metadata_str', '-s', help='submission metadata as a JSON string')
 @click.option('--data_dir', '-d', type=click.Path(exists=True),
               help='path containing submission data files')
-@click.option('--skip_md5sum_check', is_flag=True, help='skip md5sum check, save time for large files')
+#@click.option('--skip_checks', default=[],nargs=0, help='skip these tests')
+@click.option('--skip_checks','-k', multiple=True,default=[], help='skip this tests',type=click.Choice(['c685','c683','c680','c670','c609','c608']))
```
Comment (lindaxiang) [inline seq_tools/cli/__init__.py:60]: @edsu7 , do you mean `c681` instead of `c685`?

**Review by lindaxiang — APPROVED** [review]

**Pull request merged**